lib/sift.rb

```diff
@@ -26,184 +26,170 @@ class Sift
 
   class Error < StandardError; end
 
-    class Risk
-      attr_accessor :risk, :response
+  class Risk
+    attr_accessor :risk, :response
 
-      def initialize(risk:, response:, topic_hash:)
-        @risk = risk,
-        @response = response
-        @topic_hash = topic_hash
-      end
+    def initialize(risk:, response:, topic_hash:)
+      @risk = risk,
+      @response = response
+      @topic_hash = topic_hash
+    end
 
-      def over_any_max_risk
-        result = false
-        @topic_hash.each do |topic_id, risk|
-          topic_name = TopicMap[topic_id.to_i]
-          unless topic_name.nil?
-            site_setting_name = "sift_#{topic_name}_deny_level"
-            max_risk = SiteSetting.send(site_setting_name)
-            if !max_risk.nil? and risk.to_i > max_risk.to_i
-              #Rails.logger.error("sift_debug: risk greater than max")
-              return true
-            end
-          end
-        end
+    def over_any_max_risk
+      @topic_hash.detect(-> { false }) do |topic_id, risk|
```

Comment:
It might be more clear to reverse the condition so it returns true, rather than having a labmda that looks for `false`